src/instructlab/configuration.py

```diff
@@ -312,47 +333,84 @@ def fill_defaults(
 class _serve_vllm(BaseModel):
     """Class describing configuration of vllm serving backend."""
 
-    llm_family: str = ""
-    max_startup_attempts: int | None = None
-    gpus: Optional[int] = None
+    llm_family: str = Field(
+        default="",  # TODO: convert to None and use a pattern to validate
+        description="Large Language Model Family",
+        examples=["merline", "granite"],
+    )
+    max_startup_attempts: int | None = Field(
+        default=None,
+        description="Maximum number of attempts to start the vLLM server.",
+    )
+    gpus: Optional[int] = Field(default=None, description="Number of GPUs to use.")
     # arguments to pass into vllm process
-    vllm_args: list[str] | None = Field(default_factory=lambda: [])
+    vllm_args: list[str] | None = Field(
+        default_factory=list,
+        description="vLLM specific arguments. All settings can be passed as a list of strings, see: https://docs.vllm.ai/en/v0.5.2/serving/openai_compatible_server.html#command-line-arguments-for-the-server.",
```

Comment:
https://docs.vllm.ai/en/latest/serving/openai_compatible_server.html ?

Reply:
I swear last time I looked I couldn't find the startup options... thanks